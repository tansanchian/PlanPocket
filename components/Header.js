import { View, Text, TouchableOpacity, Image, StyleSheet } from "react-native";
import React from "react";
import { useNavigation } from "@react-navigation/native";
import AntDesign from "react-native-vector-icons/AntDesign";

const Header = ({ title, onPress }) => {
  const navigation = useNavigation();

  return (
    <View style={styles.container}>
      <View style={{ flexDirection: "row", alignItems: "center" }}>
        <TouchableOpacity
          onPress={() => navigation.toggleDrawer()}
          style={styles.iconContainer}
        >
          <AntDesign name="bars" size={20} />
        </TouchableOpacity>
        <Text
          style={{
            marginLeft: 12,
            fontSize: 17,
            fontWeight: "bold",
          }}
        >
          {title}
        </Text>
      </View>
      <TouchableOpacity onPress={onPress} style={styles.iconContainer}>
        <AntDesign name="ellipsis1" size={20} />
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between",
    marginTop: 20,
    marginHorizontal: 16,
  },
  iconContainer: {
    height: 45,
    width: 45,
    borderRadius: 999,
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: "#f3eef6",
  },
  icon: {
    height: 24,
    width: 24,
    tintColor: "black",
  },
});
export default Header;
