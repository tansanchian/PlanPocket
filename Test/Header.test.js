import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import { useNavigation } from "@react-navigation/native";
import Header from "../components/Header";

jest.mock("@react-navigation/native", () => ({
  useNavigation: jest.fn(),
}));

describe("Header Component", () => {
  const mockToggleDrawer = jest.fn();

  beforeEach(() => {
    useNavigation.mockReturnValue({
      toggleDrawer: mockToggleDrawer,
    });
  });

  it("should render with the correct title", () => {
    const { getByText } = render(<Header title="Test Title" />);
    expect(getByText("Test Title")).toBeTruthy();
  });

  it("should call toggleDrawer when the icon is pressed", () => {
    const { getByTestId } = render(<Header title="Test Title" />);
    fireEvent.press(getByTestId("drawer-toggle"));
    expect(mockToggleDrawer).toHaveBeenCalled();
  });

  it("should match the snapshot", () => {
    const { toJSON } = render(<Header title="Test Title" />);
    expect(toJSON()).toMatchSnapshot();
  });
});
