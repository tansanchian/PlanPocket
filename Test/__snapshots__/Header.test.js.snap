// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should match the snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginTop": 40,
      "maxHeight": 65,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#f3eef6",
          "borderRadius": 999,
          "height": 45,
          "justifyContent": "center",
          "marginHorizontal": 16,
          "opacity": 1,
          "width": 45,
        }
      }
      testID="drawer-toggle"
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": undefined,
              "fontSize": 20,
            },
            undefined,
            {
              "fontFamily": "anticon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        î˜¹
      </Text>
    </View>
    <Text
      style={
        {
          "fontSize": 17,
          "fontWeight": "bold",
          "marginLeft": 12,
        }
      }
    >
      Test Title
    </Text>
  </View>
</View>
`;
